alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/interfaces/geometry/euclidean/twod/Segment2D.kt

```diff
@@ -14,84 +16,84 @@ data class Segment2D<P : Vector2D<P>>(val first: P, val second: P) {
     /**
      * Checks whether the segment is aligned to the x axis.
      */
-    val xAxisAligned: Boolean get() = MathUtils.fuzzyEquals(first.y, second.y)
+    val xAxisAligned: Boolean by lazy { MathUtils.fuzzyEquals(first.y, second.y) }
 
     /**
      * Checks whether the segment is aligned to the y axis.
      */
-    val yAxisAligned: Boolean get() = MathUtils.fuzzyEquals(first.x, second.x)
+    val yAxisAligned: Boolean by lazy { MathUtils.fuzzyEquals(first.x, second.x) }
 
     /**
      * Checks whether the segment is axis-aligned.
      */
-    val isAxisAligned: Boolean get() = xAxisAligned || yAxisAligned
+    val isAxisAligned: Boolean by lazy { xAxisAligned || yAxisAligned }
```

Comment:
This is actually way worse than computing every time. The cost of an or operation is much smaller than a lazy delegate.